Data Structure/Linked List/LinkedListHomework1.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+class Node {
+    /** @var int */
+    private $data;
+    /** @var Node */ 
+    private $next; 
+
+    /**
+    * Constructor Node class
+    */
+    public function __construct($data = 0, $next = null)  // default value of $data is 0, $next is null
+    {
+        $this->data = $data; // initial $data
+        $this->next = $next;  // initial $next
+    }
+
+    /**
+    * get data
+    * @return int
+    */
+    public function getData()
+    {
+        return $this->data;
+    }
+
+    /**
+    * set data
+    * @param int $data
+    */
+    public function setData($data)
+    {
+        $this->data = $data;
+    }
+
+    /**
+    * get next
+    * @return Node
+    */
+    public function getNext()
+    {
+        return $this->next;
+    }
+
+    /**
+    * set next
+    * @param Node $next
+    */
+    public function setNext($next)
+    {
+        $this->next = $next;
+    }
+    
+}
+
+class LinkedList { 
+    /** @var Node  head node */
+    private $head;
+
+    /**
+    *@param int $data
+    */
+    public function insert($data)
+    {
+        $newNode = new Node($data); // create a Node
+        if ($this->head == null) {
+            // if the head is null, that mean linked list is empty, so the first node is head
+            $this->head = $newNode;
+        } else {
+            // if linked list is not null, new node will be add to end of list
+            // find the last node
+            $last = $this->head;
+            while ($last->getNext() != null) {
+                $last = $last->getNext();
+            }
+            // insert new node to at last node
+            $last->setNext($newNode);
+        }
+    }
+
+    public function deleteAll($data)
+    {
+        if ($this->head == null) { // linked list is empty
+            echo "List is empty.";
```

Comment:
Because list is empty, you should finish the function here by `return`

Reply:
yes, use `return` or need `die` 
`echo "List is empty."; die;`